.github/workflows/main.yaml

```diff
@@ -40,12 +47,13 @@ jobs:
       - run: |
           python -m pip install -e .[dev]
           python -m pip list
+
       - name: Running Tests
         run: |
           pytest --verbose --cov=. --cov-report=xml
+
       - name: Upload coverage to Codecov
-        uses: codecov/codecov-action@v5.4.0
-        if: ${{ matrix.python-version }} == 3.10
+        uses: codecov/codecov-action@5a1091511ad55cbe89839c7260b706298ca349f7 # v5.5.1
         with:
           file: ./coverage.xml
           fail_ci_if_error: false
```

Comment:
Codecov upload currently failing with `Upload failed: {"message":"Token required because branch is protected"}`

Need to fix following instructions at https://docs.codecov.com/docs/codecov-tokens#uploading-without-a-token? Do it later in a separate PR.